zerver/decorator.py

```diff
@@ -747,14 +750,13 @@ def wrapped_func(request: HttpRequest, *args: object, **kwargs: object) -> HttpR
 
             if isinstance(user, AnonymousUser) or (settings.ZILENCER_ENABLED and
                                                    isinstance(user, RemoteZulipServer)):
-                # We can only rate-limit logged-in users for now.
-                # We also only support rate-limiting authenticated
-                # views right now.
-                # TODO: implement per-IP non-authed rate limiting
+                ip_addr = request.META["REMOTE_ADDR"]
+                assert ip_addr
+                rate_limit_ip(request, ip_addr, domain='api_by_ip')
```

Comment:
We're using the `SetRemoteAddrFromForwardedFor` middleware, in conjunction with the nginx configuration that clears/sets that value so it can't be provided by the client, to know that this is the actual end-user IP and not any proxies.  That seems like a good thing to insert into a comment here.